### PR TITLE
fix verify crash caused by invalid cert format

### PR DESCRIPTION
**Pull request by qile222**
function verify should return after call Nan::ThrowError, otherwise point memory will be free multi times, and then crash the main thread.

### CODE REVIEW

**Comment by yorkie** [comment]
Hi @lolBig I have fixed the verification expiration issue at #62, and may I ask you to add some test for your patch?

File: test/test.js [hunk 1]
```diff
@@ -38,3 +38,19 @@ x509.verify(
     assert.throws(assert.ifError.bind(null, err), /ENOENT/)
   }
 );
+
+x509.verify(
+  path.join(__dirname, 'certs/equifax.crt'),
+  path.join(__dirname, '/test.js'),
+  function(err, result) {
+    assert.notStrictEqual(err, null);
```
Comment (yorkie) [inline test/test.js]: Should throw an invalid format error here?

Comment (lolBig) [reply test/test.js]: error was caught by function verify

**Comment by yorkie** [comment]
Rad! LGTM if CI is green @lolBig :)

**Comment by yorkie** [comment]
Merged and published at v0.3.3 :)

/cc @Southern

**Comment by Southern** [comment]
Why is there a `do {...} while (0);` loop introduced in this code?

**Pull request merged**